sql/functions/show_partitions.sql

```diff
@@ -86,11 +90,10 @@ END IF;
 v_sql := v_sql || format('
     AND pg_get_expr(relpartbound, c.oid) != ''DEFAULT'' ');
 
-
-IF v_control_type = 'time' THEN
+IF v_control_type = 'time' OR (v_control_type IN ('text', 'uuid')) THEN
 
     v_sql := v_sql || format('
-        ORDER BY (regexp_match(pg_get_expr(c.relpartbound, c.oid, true), $REGEX$\(([^)]+)\) TO \(([^)]+)\)$REGEX$))[1]::text::timestamptz %s '
+        ORDER BY (regexp_match(pg_get_expr(c.relpartbound, c.oid, true), $REGEX$\(([^)]+)\) TO \(([^)]+)\)$REGEX$))[1] %s '
```

Comment:
This has to convert to a timestamp to get the expected logical order out no? Otherwise doesn't this sort by text ordering?

Reply:
The intention of this function is to show them in the logical, decoded time order. Otherwise the other functions that use this function to get the list of child tables for things like retention may act on the wrong child tables.

Reply:
Understood. I was thinking for text column partitioning, one would always use lexicographically sorted identifiers like uuidv7/ulid, so this query should implicitly be time ordered.
Do you think it's safer to explicitly convert to timestamp and sort?

Reply:
I'd like to keep it consistently time ordered for all time-based partitioning. So I think it would be safer to convert it here.

Reply:
Done, have pushed a new commit with this change